Feat/fetch fetch from API

D3nnis249 opened the pull request:
Hello and thank you for your review. In this PR, we have exchanged our fake data for an Api fetch and added this fetch with a radius filter.

D3nnis249 committed:
Fetch Stations, get location from user

Co-authored-by: AlexF090
<AlexF090@users.noreply.github.com>

AlexF090 committed:
add fetch data from the api with radius

Co-authored-by: D3nnis249 <D3nnis249@users.noreply.github.com>

D3nnis249 committed:
fix for vercel

Co-authored-by: AlexF090
<AlexF090@users.noreply.github.com>

mbosselmann commented on client/src/App.jsx:
```diff
@@ -8,14 +8,34 @@ import SettingsPage from '../src/pages/SettingsPage';
+  const url = `https://api.tankentanken.de/gas-stations?latitude=${latitude}&longitude=${longitude}&radius=${apiRadius}`;
   const FuelInfo = fuelValue.charAt(0).toUpperCase() + fuelValue.slice(1);
 
+  //Calculate radius for API
```
I think that these comments aren't really necessary as the names of your functions are good. Therefore, I would recommend you to remove them.

mbosselmann commented on client/src/App.jsx:
```diff
@@ -8,14 +8,34 @@ import SettingsPage from '../src/pages/SettingsPage';
   const FuelInfo = fuelValue.charAt(0).toUpperCase() + fuelValue.slice(1);
 
+  //Calculate radius for API
+  function calculateApiRadius(radius) {
```
I think as name `calculateRadius` would be fine as this function could also be used for other purposes. :)

mbosselmann commented on client/src/App.jsx:
```diff
@@ -25,36 +45,86 @@ function App() {
+      },
+    })
+      .then(response => response.json())
+      .then(json => setStations(json));
```
I would suggest you to add some error handling with `catch`

D3nnis249 replied:
Was discussed with Mareike 👍

mbosselmann commented on client/src/App.jsx:
```diff
@@ -8,14 +8,34 @@ import SettingsPage from '../src/pages/SettingsPage';
   );
 
+  const url = `https://api.tankentanken.de/gas-stations?latitude=${latitude}&longitude=${longitude}&radius=${apiRadius}`;
   const FuelInfo = fuelValue.charAt(0).toUpperCase() + fuelValue.slice(1);
```
As in React components names begin with a capital letter, you should not name this `const` `FuelInfo`, but `fuelInfo`.

D3nnis249 replied:
Ups, thanks :D

mbosselmann commented on client/src/components/Button.jsx:
```diff
@@ -0,0 +1,7 @@
+  return <button onClick={myFunction}>{buttonTitle}</button>;
+}
+
+export default Button;
```
I like! 👍 But no stylings for this button? 🤔

D3nnis249 replied:
We first took care of the functionality, and today we take care of the styling. 👍

mbosselmann reviewed (commented):
You rock! 🪨 Left some remarks for you to consider. :)

AlexF090 committed:
favorites page broken this commit is for the coaches

Co-authored-by: D3nnis249 <D3nnis249@users.noreply.github.com>

AlexF090 committed:
Merge branch 'feat/fetch' of github.com:D3nnis249/GenialTanken-Capstone into feat/fetch

AlexF090 committed:
add favorite fetch with console.log

Co-authored-by: D3nnis249
<D3nnis249@users.noreply.github.com>

AlexF090 committed:
display favorite stations on FavoritePage

Co-authored-by: D3nnis249
<D3nnis249@users.noreply.github.com>

AlexF090 committed:
add fetch to DetailPage

Co-authored-by: D3nnis249
<D3nnis249@users.noreply.github.com>

AlexF090 committed:
add Price component

and add the price component to the GSList GSDetailList and Map

Co-authored-by: D3nnis249
<D3nnis249@users.noreply.github.com>

AlexF090 committed:
add Price component to GSDetailList

and add catch error in all prices

Co-authored-by: D3nnis249
<D3nnis249@users.noreply.github.com>

AlexF090 committed:
fix misstake in GSDetaiList in "Diesel" price

Co-authored-by: D3nnis249
<D3nnis249@users.noreply.github.com>

D3nnis249 committed:
change Header Tag

Co-authored-by: AlexF090 <AlexF090@users.noreply.github.com>

D3nnis249 committed:
add distance to stationcard and css

Co-authored-by: AlexF090 <AlexF090@users.noreply.github.com>

D3nnis249 committed:
fixed issues PR 33

Co-authored-by: AlexF090
<AlexF090@users.noreply.github.com>

mbosselmann commented on client/src/components/GasStationList.jsx:
```diff
@@ -77,13 +89,17 @@ const CustomLink = styled(Link)`
   align-self: center;
-  margin-right: 1em;
+  font-weight: 600;
+  /* margin-right: 1em; */
```
Remove commented code please ;)

mbosselmann commented on client/src/components/GasStationList.jsx:
```diff
@@ -99,18 +115,26 @@ const Adress = styled.p`
+  justify-self: end;
+  align-self: end;
+  font-weight: 600;
+  /* align-self: center; */
```
Same here

mbosselmann reviewed (approved):
Approved. 🚀 Just two mini comments.

D3nnis249 committed:
Fix for PR 33

Co-authored-by: AlexF090 <AlexF090@users.noreply.github.com>
